dist/tools/usb-cdc-ecm/start_network.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+
+USB_CDC_ECM_DIR="$(dirname $(readlink -f $0))"
+
+INTERFACE="enp0s20f0u1"
+
+setup_interface() {
+    sysctl -w net.ipv6.conf.${INTERFACE}.forwarding=1
+    sysctl -w net.ipv6.conf.${INTERFACE}.accept_ra=0
+    ip link set ${INTERFACE} up
+    ip a a fe80::1/64 dev ${INTERFACE}
+    ip a a fd00:dead:beef::1/128 dev lo
+    ip route add ${PREFIX} via fe80::2 dev ${INTERFACE}
+}
+
+cleanup_interface() {
+    ip a d fe80::1/64 dev ${INTERFACE}
+    ip a d fd00:dead:beef::1/128 dev lo
+    ip route del ${PREFIX} via fe80::2 dev ${INTERFACE}
+}
+
+cleanup() {
+    echo "Cleaning up..."
+    cleanup_interface
+    if [ -n "${UHCPD_PID}" ]; then
+        kill ${UHCPD_PID}
+    fi
+    if [ -n "${DHCPD_PIDFILE}" ]; then
+        kill "$(cat ${DHCPD_PIDFILE})"
+        rm "${DHCPD_PIDFILE}"
+    fi
+    trap "" INT QUIT TERM EXIT
+}
+
+start_uhcpd() {
+    ${UHCPD} ${INTERFACE} ${PREFIX} > /dev/null &
+    UHCPD_PID=$!
+}
+
+start_dhcpd() {
+    DHCPD_PIDFILE=$(mktemp)
+    ${DHCPD} -d -p ${DHCPD_PIDFILE} ${INTERFACE} ${PREFIX} 2> /dev/null
+}
+
+if [ "$1" = "-d" ] || [ "$1" = "--use-dhcpv6" ]; then
+    USE_DHCPV6=1
+    shift 1
+else
+    USE_DHCPV6=0
+fi
+
+PREFIX=$1
+[ -z "${PREFIX}" ] && {
+    echo "usage: $0 [-d|--use-dhcpv6] <prefix> "
+    exit 1
+}
+
+trap "cleanup" INT QUIT TERM EXIT
+
+setup_interface
+
+if [ ${USE_DHCPV6} -eq 1 ]; then
+    DHCPD="$(readlink -f "${USB_CDC_ECM_DIR}/../dhcpv6-pd_ia/")/dhcpv6-pd_ia.py"
+    start_dhcpd
+else
+    UHCPD="$(readlink -f "${USB_CDC_ECM_DIR}/../uhcpd/bin")/uhcpd"
+    start_uhcpd
+fi
```

Comment:
As soon as this part is reached, the `cleanup` function will be called.
When I ran 

```
% make BOARD=nrf52840dongle UPLINK=usb-cdc-ecm  term
make -C ethos
make -C uhcpd
make -C sliptty
make[2]: Nothing to be done for 'all'.
make[2]: Nothing to be done for 'all'.
make[2]: Nothing to be done for 'all'.
sudo sh /home/benpicco/dev/RIOT/dist/tools/usb-cdc-ecm/start_network.sh  2001:db8::/64
net.ipv6.conf.enx66410e90125b.forwarding = 1
net.ipv6.conf.enx66410e90125b.accept_ra = 0
Cleaning up...
```

was all I get - and 

```
2020-07-15 16:02:31,407 # uhcp_client(): sending REQ...
2020-07-15 16:02:41,406 # uhcp_client(): no reply received
2020-07-15 16:02:41,407 # uhcp_client(): sending REQ...
2020-07-15 16:02:51,406 # uhcp_client(): no reply received
```

on the device.

When I added a `sleep 10` here, it would work - but that's a hack.

```
2020-07-15 16:05:51,414 # got packet from fe80::1 port 48615
2020-07-15 16:05:51,415 # uhcp: push from fe80::1:48615 prefix=2001:db8::/64
2020-07-15 16:05:51,417 # gnrc_uhcpc: uhcp_handle_prefix(): add compression context 0 for prefix 2001:db8::ac0f:e90:125b:e86d/64
2020-07-15 16:05:51,420 # gnrc_uhcpc: uhcp_handle_prefix(): configured new prefix 2001:db8::ac0f:e90:125b:e86d/64
```

Also this claims to be `TERMPROG`, but it doesn't launch a terminal.
So why not kill two birds with one stone and launch `pyterm` here?

Still a bit of a hack, but that would get the desired result.

Reply:
859eba150

Added an optional argument to `start_network.sh` so that a serial port can be specified and opened using `pyterm`. This is passed from `$PORT` in the Makefile, so that it always opens for the example. I specifically want this to be optional for "Headless" deployments where only the ethernet interface is connected, so we can just to `start_network $PREFIX` for those.

Hadn't realised that I'd removed the wait for key from this before submitting, as the `EXIT` does trigger the cleanup `trap`.

TLDR; A bit of a hack, but not the end of the world.